### PR TITLE
Add cmp/sym for Arduino Nano v3

### PR DESCRIPTION
**Pull request by dbrgn**
Symbol:

img

Component:

img

### CODE REVIEW

File: sym/039a25f3-c54b-4ab5-8c27-737e5cb5a491/symbol.lp [hunk 1]
```diff
@@ -0,0 +1,108 @@
+(librepcb_symbol 039a25f3-c54b-4ab5-8c27-737e5cb5a491
+ (name "Arduino Nano v3")
+ (description "A cheap Arduino board based on the ATMEGA328P.")
+ (keywords "arduino,nano,atmega328p")
+ (author "Danilo Bargen")
+ (version "0.1")
+ (created 2018-11-17T11:47:05Z)
+ (deprecated false)
+ (category e96d0dda-3100-4e68-8c33-e2a5bc80428b)
+ (pin cf653412-dc0b-465a-ac7d-912006615400 (name "D1/TX")
```
Comment (ubruhin) [inline sym/039a25f3-c54b-4ab5-8c27-737e5cb5a491/symbol.lp]:
```suggestion
 (pin cf653412-dc0b-465a-ac7d-912006615400 (name "D1")
```
For consistency, see for example https://github.com/LibrePCB-Libraries/Microchip.lplib/pull/2.

File: sym/039a25f3-c54b-4ab5-8c27-737e5cb5a491/symbol.lp [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  (position 12.7 15.24) (rotation 180.0) (length 2.54)
+ )
+ (pin 8846654d-43c8-4d1f-a085-789af061506e (name "D0/RX")
```
Comment (ubruhin) [inline sym/039a25f3-c54b-4ab5-8c27-737e5cb5a491/symbol.lp]:
```suggestion
 (pin 8846654d-43c8-4d1f-a085-789af061506e (name "D0")
```
For consistency, see for example https://github.com/LibrePCB-Libraries/Microchip.lplib/pull/2.

File: cmp/7f8cadda-e30b-43fc-87c4-04a9fe0f70d2/component.lp [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+ (name "Arduino Nano v3")
+ (description "A cheap Arduino board based on the ATMEGA328P.")
+ (keywords "arduino,nano,atmega328p")
+ (author "Danilo Bargen")
+ (version "0.1")
+ (created 2018-11-17T12:05:35Z)
+ (deprecated false)
+ (category e96d0dda-3100-4e68-8c33-e2a5bc80428b)
+ (schematic_only false)
+ (default_value "{{PARTNUMBER or DEVICE}}")
```
Comment (ubruhin) [inline cmp/7f8cadda-e30b-43fc-87c4-04a9fe0f70d2/component.lp]:
```suggestion
 (default_value "{{PARTNUMBER or DEVICE or COMPONENT}}")
```
With that you can see "Arduino Nano v3" in schematics even if the device is not placed yet on the board.

File: sym/039a25f3-c54b-4ab5-8c27-737e5cb5a491/symbol.lp (hunk 1 above)
Comment (dbrgn) [reply sym/039a25f3-c54b-4ab5-8c27-737e5cb5a491/symbol.lp]: The pin is actually called TX1 on the silkscreen (and D0/RX is labelled RX0), so I though putting both there makes sense in that case. No other secondary function is labelled on the silkscreen.

See here, bottom left:

img

What do you think?

File: cmp/7f8cadda-e30b-43fc-87c4-04a9fe0f70d2/component.lp (hunk 3 above)
Comment (dbrgn) [reply cmp/7f8cadda-e30b-43fc-87c4-04a9fe0f70d2/component.lp]: Ahh, good point!

**Comment by dbrgn** [comment]
> Left-align name and value

It would be good to add some clear guidelines for that. I still think for small symbols (think resistors, transistors, etc) it makes sense to center the name/value, because for wide labels the width is shared on both sides. You're free to disagree of course :)

On wide symbols, it doesn't really matter to me. Might look better left-aligned.

Should we choose a guideline, e.g. "center for symbols less than 8*2.54mm wide, left-align for wider symbols", or stick with one approach for consistency?

**Comment by ubruhin** [comment]
> I still think for small symbols (think resistors, transistors, etc) it makes sense to center the name/value, because for wide labels the width is shared on both sides.

I guess we had this discussion for footprints, but footprints != symbols so we can have different guidelines for them ;) For symbols I think left-align should always look good for "IC-like" symbols, no matter how large they are. But for other kinds of symbols it's *very* hard to define guidelines as their shapes are very different...

File: sym/039a25f3-c54b-4ab5-8c27-737e5cb5a491/symbol.lp (hunk 1 above)
Comment (ubruhin) [reply sym/039a25f3-c54b-4ab5-8c27-737e5cb5a491/symbol.lp]: Ah, good point. Then it's OK for me :)

**Comment by dbrgn** [comment]
> For symbols I think left-align should always look good for "IC-like" symbols, no matter how large they are.

Would be great if you could add that to https://github.com/LibrePCB/librepcb-doc/pull/18 :slightly_smiling_face:

**Comment by dbrgn** [comment]
On second thought, I think it's confusing for users/contributors (especially new contributors) if guidelines are different for symbols and packages. I'll add a discussion item in the librepcb-doc repository.

**Comment by ubruhin** [comment]
> On second thought, I think it's confusing for users/contributors (especially new contributors) if guidelines are different for symbols and packages. I'll add a discussion item in the librepcb-doc repository.

Could you still go with left aligned text until this is clarified? Then it's at least consistent across all of our libraries.

Would be good to get this merged soon, currently the library manager downloads a completely empty library which is a bit annoying ;)

**Comment by dbrgn** [comment]
@ubruhin done

**Comment by ubruhin** [comment]
Did you saw my comment above?

> Is there a special reason for the order of the pins at bottom left? Generally I would expect VCC and GND to be next to each other, but currently AREF and RST are in between. Also I would sort VCC pins by voltage, i.e. VIN > 5V > 3V3 :)

**Comment by dbrgn** [comment]
Nope, I overlooked that :slightly_smiling_face: I put GND at the bottom so that I can easily connect the GND symbol towards the bottom.

How about this?

img

**Comment by ubruhin** [comment]
Thanks :) I would say that looks good :+1:

**Pull request merged**